bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/device/Device.java

```diff
@@ -0,0 +1,151 @@
+    public Device(String[] items) {
+        dataItems = items;
+    }
+
+    /**
+     * Return the data item at the given index
+     *
+     * @return A String containing the data item at the Index or "" if the Index is invalid
+     */
+    public String getDataItem(int index) {
+        if (index >= 0 && index < dataItems.length) {
+            return dataItems[index];
+        } else {
+            logger.warn("getDataItem: Invalid Index {}.", index);
+            return "";
+        }
+    }
+
+    /**
+     * @param dataItems
+     */
+    public void setDataItems(String[] dataItems) {
+        this.dataItems = dataItems;
+    }
+
+    /**
+     * Returns the value of the dataItem at Index as a Float
+     *
+     * @param index Index of the dataItem to return
+     * @return Float or null if the index is invalid or the dataItem can not be converted to Float
+     */
+    public @Nullable Float getFloat(int index) {
+        Float result = null;
+
+        if (index >= 0 && index < dataItems.length) {
+            try {
+                result = Float.valueOf(dataItems[index]);
+            } catch (NumberFormatException e) {
+                logger.warn("getFloat: {}.", e.getMessage());
+            }
+        } else {
+            logger.warn("getFloat: Invalid Index {}.", index);
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns the value of the dataItem at Index as a String
+     *
+     * @param index Index of the dataItem to return
+     * @return String or null if the index is invalid
+     */
+    public @Nullable String getString(int index) {
+        if (index >= 0 && index >= dataItems.length) {
+            logger.warn("getString: Invalid Index {}.", index);
+            return null;
+        }
+
+        return dataItems[index];
+    }
+
+    /**
+     * Returns the value of the dataItem at Index as a Integer
+     *
+     * @param index Index of the dataItem to return
+     * @return Integer or null if the index is invalid or the dataItem can not be converted to Integer
+     */
+    public @Nullable Integer getInteger(int index) {
+        Integer result = null;
+
+        if (index >= 0 && index < dataItems.length) {
+            try {
+                result = Integer.valueOf(dataItems[index]);
+            } catch (NumberFormatException e) {
+                logger.warn("getInteger: {}.", e.getMessage());
+            }
+        } else {
+            logger.warn("getInteger: Invalid Index {}.", index);
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns the value of the dataItem at Index as a Boolean
+     *
+     * @param index Index of the dataItem to return
+     * @return true if the first char if 1, false if the first char is 0 or null if the index is invalid or the dataItem
+     *         can not be converted to Boolean
+     */
+    public @Nullable Boolean getBoolean(int index) {
+        Boolean result = null;
+
+        if (index >= 0 && index < dataItems.length) {
```

Comment:
And also here

Reply:
I am really not insisting to keep it like this but if i change it according to your suggestion, i would have to either call getDataItem twice for both if's (which does not feel "nice" to me) or call it once and store the result in a String to use in the if's which is OK for me but not really an improvement other than bringing cyclomatic complexity down, but it is low here already anyway... I am ambivalent - if you want it changed, i'll change it.